### PR TITLE
Unit test focusing on hard-coded constants

### PR DESCRIPTION
**Pull request by hugovdm**
Blocked on https://github.com/sffc/icu/pull/48 which will open up some internals for more direct testing.

##### Checklist

- [ ] Issue filed: https://unicode-org.atlassian.net/browse/ICU-_____
- [ ] Updated PR title and link in previous line to include Issue number
- [ ] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: docs/userguide/format_parse/numbers/measurement_units.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Measurement Units
+
+TODO(hugovdm,younies): this is a work in progress. This document should be
+cleaned up before merging into the upstream unicode-org ICU repository. Some of
+this content belongs in header files.
+
+## Internal Implementation Details: ICU4C
```
Comment (sffc) [inline docs/userguide/format_parse/numbers/measurement_units.md]: Implementation docs should go as comments in the header or source files.

File: icu4c/source/test/intltest/unitstest.cpp [hunk 2]
```diff
@@ -52,6 +56,75 @@ void UnitsTest::runIndexedTest(int32_t index, UBool exec, const char *&name, cha
     TESTCASE_AUTO_END;
 }
 
+// Tests freshness of the constants in units.txt by simply loading each directly
+// specified conversion rate into UnitConverter.
+void UnitsTest::testConstantFreshness() {
```
Comment (sffc) [inline icu4c/source/test/intltest/unitstest.cpp]: What are you testing?  There are no assertions in this function.

**Review by sffc — CHANGES_REQUESTED** [review]

File: docs/userguide/format_parse/numbers/measurement_units.md (hunk 1 above)
Comment (hugovdm) [reply docs/userguide/format_parse/numbers/measurement_units.md]: I'm moving this to a Google Doc - much easier to comment/discuss, and intended as a way to help us better structure our approach to ICU4J.

File: icu4c/source/test/intltest/unitstest.cpp (hunk 2 above)
Comment (hugovdm) [reply icu4c/source/test/intltest/unitstest.cpp]: Example output when I change the "acre to meter-squared" definition to use a non-existent "acore_to_m2" constant:

         testConstantFreshness data: expected success but got error: U_INVALID_FORMAT_ERROR - UnitConverter(<acre>, <square-meter>, status).
         If U_INVALID_FORMAT_ERROR, please check that "icu4c/source/i18n/unitconverter.cpp" has all constants?
         Factor for "acre" is: "acre_to_m2".
         Full list of constants: "G" "PI" "ft2_to_m2" "ft3_to_m3" "ft_to_m" "gal_imp_to_m3" "gal_to_m3" "gravity" "in3_to_m3" "lb_to_kg" - (Are you missing data?)

**Comment by hugovdm** [comment]
Closing this PR in favour of https://github.com/icu-units/icu/pull/20 - its first commit (678b3d5) is a squashed version of this PR, rebased on units-staging.

**Pull request closed without merging**